impl/client/src/main/javascript/web/pentaho/type/property.js

```diff
@@ -280,6 +281,47 @@ define([
         },
         // endregion
 
+        // region alias attribute
+        _nameAlias: undefined,
+
+        /**
+         * Gets or sets the alias of the _property type_.
+         *
+         * The alias of a _property type_ is an alternative identifier for serialization purposes.
+         *
+         * ### Set
+         *
+         * This attribute must be set when defining a new _property type_,
+         * and cannot be changed afterwards.
+         *
+         * When set to a non-{@link Nully} and non-{@link String} value,
+         * the value is first replaced by the result of calling its `toString` method.
+         *
+         * @type {!nonEmptyString}
+         *
+         * @throws {pentaho.lang.ArgumentRequiredError} When set to an empty string or a _nully_ value.
+         * @throws {TypeError} When set to a value different from the current one.
+         *
+         * @see pentaho.type.spec.IPropertyTypeProto#nameAlias
+         */
+        get nameAlias() {
+          return this._nameAlias;
+        },
+
+        set nameAlias(value) {
+          value = nonEmptyString(value);
+
+          if(!value) throw error.argRequired("nameAlias");
+
+          // Hierarchy consistency
+          var baseValue = this._nameAlias;
+          if(baseValue && value !== baseValue)
```

Comment:
The name of a property is settled on the _root_ property type `R` and cannot change in sub property types, `A` and `B`.
Name aliases, however, can be set on the root or only on subtrees, and still cannot change in that subtree once set. This means that a serialization of a type `A` is not necessarily understood by type `B`, even though a property is defined in `R`. Given a spec of `B`, one must first name aliases into actual names of `R` and only then can the spec be correctly read by `A`. I think this is an acceptable limitation, but if you guys think this brings undesirable concerns at the cost of a neglectable increase of functionality, then we can force the alias to only be settable at the root property.